### PR TITLE
New package: sparse-0.5.2

### PR DESCRIPTION
**Pull request by ailiop-git**

### CODE REVIEW

File: srcpkgs/sparse/template [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Template file for 'sparse'
+pkgname=sparse
+version=0.5.2
+revision=1
+build_style=gnu-makefile
+make_use_env=no
```
Comment (abenson) [inline srcpkgs/sparse/template]: The default is to not use environment -- by setting this string to anything you make it use environment.

File: srcpkgs/sparse/template [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+short_desc="Sparse - a Semantic Parser for C"
```
Comment (abenson) [inline srcpkgs/sparse/template]: Shorten this to "Semantic parser for C"

File: srcpkgs/sparse/template (hunk 1 above)
Comment (newbluemoon) [reply srcpkgs/sparse/template]: Just to elaborate a bit more: `common/build-style/gnu-makefile.sh` tests for

    if [ -z "$make_use_env" ]; then

So it only matters if it is empty or not. The actual value if set has no meaning.

**Comment by ailiop-git** [comment]
thank you for the reviews, all comments are now addressed.

File: srcpkgs/sparse/template [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+short_desc="Semantic Parser for C"
+maintainer="Anthony Iliopoulos <ailiop@altatus.com>"
+license="MIT"
+homepage="https://git.kernel.org/pub/scm/devel/sparse/sparse.git"
+distfiles="${homepage}/snapshot/${pkgname}-${version}.tar.gz"
+checksum=5ca5b7be7240a97fbdc6194f18d64837f9af070aeb0c3052f0a5d014acf1d048
+
+if [ -n "$CROSS_BUILD" ]; then
+	make_build_args+="
+		CC=${XBPS_CROSS_TRIPLET}-gcc \
```
Comment (Johnnynator) [inline srcpkgs/sparse/template]: `CC=$CC LD=$LD AR=$AR` should be enough

Comment (ailiop-git) [reply srcpkgs/sparse/template]: done, thanks for the feedback.

File: srcpkgs/sparse/template [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+make_use_env=yes
+short_desc="Semantic Parser for C"
+maintainer="Anthony Iliopoulos <ailiop@altatus.com>"
+license="MIT"
+homepage="https://git.kernel.org/pub/scm/devel/sparse/sparse.git"
+distfiles="${homepage}/snapshot/${pkgname}-${version}.tar.gz"
+checksum=5ca5b7be7240a97fbdc6194f18d64837f9af070aeb0c3052f0a5d014acf1d048
+
+pre_build() {
+	make_build_args="CC=$CC LD=$CC AR=$AR"
```
Comment (travankor) [inline srcpkgs/sparse/template]: `make_build_args` doesn't need to go inside `pre_build()` (it usually goes below `build_style`)

Comment (newbluemoon) [reply srcpkgs/sparse/template]: But when you put `make_build_args` in the main template body `$CC`, etc. haven’t been set, yet.

Comment (abenson) [reply srcpkgs/sparse/template]: They should be, based on $XBPS_CROSS_BUILD

Comment (ailiop-git) [reply srcpkgs/sparse/template]: Indeed the toolchain variables are set by `setup_pkg()` after the template is sourced, so for the particular usage of `make_build_args` it needs to go into the `pre_build()` step. 

Alternatively it could be set within `do_build()`, but since there is no need to override any other invocation params of `make` it looks cleaner inside `pre_build()`.

Comment (newbluemoon) [reply srcpkgs/sparse/template]: No, this case is special, because it uses the `$CC` and `$AR` variables. xbps-src sets them after `make_build_args` is read in if it was in the main template body. So these variables would be empty.

Comment (travankor) [reply srcpkgs/sparse/template]: Oh, I see, sorry for all the noise.
